File: SubtractFraction.js
numerators = [1, 1, 1, 1];//enter your own numerators
denominators = [4, 4, 4, 4];//enter your own denominators

function doSubtract() {
    canonizeFraction();

    // subtract all transformed numerators
    // STEP 3:
    answer = new_numerators[0];
    for (var i = 1; i < new_numerators.length; i++) {
        answer -= new_numerators[i];
    }
}

File: SubtractFraction.py
from AddFraction import PlusFraction

# A class with a parent
class MinusFraction(PlusFraction):

    # A constructor
    def __init__(self, fractions):
        super().__init__(fractions)


    # Returns a dictionary of the new fraction
    def doSubtract(self):
		# STEPS 1, 2:
        self.canonizeFraction()

		# STEP 3:
		# subtract all transformed numerators
        self.answer = self.new_numerators[0]
        for count in range(1, len(self.new_numerators)): self.answer -= self.new_numerators[count]

        return {'numerator':self.answer, 'denominator':self.lcm}

